<launch>
    <!-- 인자 설정 -->
    <arg name="pcd_file" default="/home/sh/catkin_ws/pcd/lab.pcd" />
    <arg name="rviz" default="true" />
    <arg name="filter_limit_min" default="0.2" />
    <arg name="filter_limit_max" default="1.5" />
    <arg name="z_threshold" default="1.0" />

    <!-- 노드에 파라미터 전달 -->
    <param name="pcd_file" value="$(arg pcd_file)" />
    <param name="z_threshold" value="$(arg z_threshold)" />

    <node pkg="pcl_ros" type="pcd_to_pointcloud" name="spawn_pcd_to_pcl" output="screen" args="$(arg pcd_file) 0.1">
        <param name="frame_id" value="map" />
        <param name="resolution" value="0.05" />
        <remap from="cloud_pcd" to="cloud_in" />
    </node>

    <node pkg="cvt_pcd2octomap" type="filter_z" name="filter_z" output="screen">
        <param name="filter_limit_min" value="$(arg filter_limit_min)" />
        <param name="filter_limit_max" value="$(arg filter_limit_max)" />
    </node>

    <node pkg="cvt_pcd2octomap" name="pcd_divider_node" type="pcd_divider_node" output="screen">
        <param name="pcd_file" value="$(arg pcd_file)" />
        <param name="z_threshold" value="$(arg z_threshold)" />
    </node>
    
    <node pkg="octomap_server" type="octomap_tracking_server_node" name="octomap_tracking_server_node" output="screen">
        <remap from="cloud_in" to="/filtered_cloud" />
        <remap from="projected_map" to="map" />
    </node>

    <node pkg="map_server" type="map_saver" name="map_saver" args="-f /home/sh/catkin_ws/maps/filtered_map_street" output="screen"/>

    <group if="$(arg rviz)">
        <node pkg="rviz" type="rviz" name="cvt_pcd_rviz" args="-d $(find cvt_pcd2octomap)/rviz/pcd2octo.rviz" />
    </group>
</launch>
